working-with-fhir-igs/hl7v2-to-dk-medcom/service.bal

```diff
@@ -0,0 +1,62 @@
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+import ballerinax/health.fhir.r4;
+import ballerinax/health.hl7v2;
+import ballerinax/health.hl7v2.utils.v2tofhirr4;
+
+type Hl7Message readonly & record {|
+    string content;
+|};
+
+configurable int servicePort = 9090;
+
+function init(){
+    log:printInfo("HL7v2 to Danish Transformation Service Started..");
+}
+
+service /hl7 on new http:Listener(servicePort) {
+    resource function post parse/danish(@http:Payload string payload) returns json|string|error {
+        string incomingMessage = payload;
+
+        // Log the incoming raw payload
+        log:printInfo("Received HL7 message", rawMessage = incomingMessage);
+
+        // Process the message considering newline characters
+        string processedMessage = "";
+        string[] segments = re `\r`.split(incomingMessage);
+        foreach string segment in segments {
+            if segment != "" {
+                processedMessage = string:'join("\n", processedMessage, segment);
+            }
+        }
```

Comment:
Can't we just pass the hl7 string as it is?

Reply:
added this part just for logging. Removed hence no functional requirement.